src/map/unit.c

```diff
@@ -157,328 +171,315 @@ static int unit_walktoxy_sub(struct block_list *bl)
 		mob->unlocktarget(md, tick);
-		if (!(++ud->walk_count%WALK_SKILL_INTERVAL))
+		if ((++ud->walk_count % WALK_SKILL_INTERVAL) != 0)
 			mob->skill_use(md, tick, -1);
-		return 0;
+		return 1;
 	}
 
 	//Refresh view for all those we lose sight
-	map->foreachinmovearea(clif->outsight, bl, AREA_SIZE, dx, dy, sd?BL_ALL:BL_PC, bl);
+	map->foreachinmovearea(clif->outsight, bl, AREA_SIZE, dx, dy, (sd != NULL ? BL_ALL : BL_PC), bl);
 
 	x += dx;
 	y += dy;
 	map->moveblock(bl, x, y, tick);
-	ud->walk_count++; //walked cell counter, to be used for walk-triggered skills. [Skotlex]
+	ud->walk_count++; // walked cell counter, to be used for walk-triggered skills. [Skotlex]
 	status_change_end(bl, SC_ROLLINGCUTTER, INVALID_TIMER); //If you move, you lose your counters. [malufett]
 
 	if (bl->x != x || bl->y != y || ud->walktimer != INVALID_TIMER)
-		return 0; //map->moveblock has altered the object beyond what we expected (moved/warped it)
+		return 1; // map->moveblock has altered the object beyond what we expected (moved/warped it)
 
 	ud->walktimer = -2; // arbitrary non-INVALID_TIMER value to make the clif code send walking packets
-	map->foreachinmovearea(clif->insight, bl, AREA_SIZE, -dx, -dy, sd?BL_ALL:BL_PC, bl);
+	map->foreachinmovearea(clif->insight, bl, AREA_SIZE, -dx, -dy, (sd != NULL ? BL_ALL : BL_PC), bl);
 	ud->walktimer = INVALID_TIMER;
 
-	if(sd) {
-		if( sd->touching_id )
-			npc->touchnext_areanpc(sd,false);
+	if (sd != NULL) {
+		if (sd->touching_id != 0)
+			npc->touchnext_areanpc(sd, false);
 		if (map->getcell(bl->m, bl, x, y, CELL_CHKNPC)) {
-			npc->touch_areanpc(sd,bl->m,x,y);
-			if (bl->prev == NULL) //Script could have warped char, abort remaining of the function.
-				return 0;
-		} else
+			npc->touch_areanpc(sd, bl->m, x, y);
+			nullpo_retr(1, bl->prev); //Script could have warped char, abort remaining of the function.
+		} else {
 			npc->untouch_areanpc(sd, bl->m, x, y);
-
-		if( sd->md ) { // mercenary should be warped after being 3 seconds too far from the master [greenbox]
-			if( !check_distance_bl(&sd->bl, &sd->md->bl, MAX_MER_DISTANCE) ) {
-				if (sd->md->masterteleport_timer == 0)
-					sd->md->masterteleport_timer = timer->gettick();
-				else if (DIFF_TICK(timer->gettick(), sd->md->masterteleport_timer) > 3000) {
-					sd->md->masterteleport_timer = 0;
-					unit->warp( &sd->md->bl, sd->bl.m, sd->bl.x, sd->bl.y, CLR_TELEPORT );
-				}
-			} else // reset the tick, he is not far anymore
-				sd->md->masterteleport_timer = 0;
-		}
-		if( sd->hd ) {
-			if( homun_alive(sd->hd) && !check_distance_bl(&sd->bl, &sd->hd->bl, MAX_MER_DISTANCE) ) {
-				if (sd->hd->masterteleport_timer == 0)
-					sd->hd->masterteleport_timer = timer->gettick();
-				else if (DIFF_TICK(timer->gettick(), sd->hd->masterteleport_timer) > 3000) {
-					sd->hd->masterteleport_timer = 0;
-					unit->warp( &sd->hd->bl, sd->bl.m, sd->bl.x, sd->bl.y, CLR_TELEPORT );
-				}
-			} else
-				sd->hd->masterteleport_timer = 0;
```

Comment:
why have all the previous behavior and checks removed?

Reply:
They've been moved to the added function unit->warpto_master since they were identical.

Reply:
Argh, sorry i did not notice this. blame my sleepy brain, disregard rest of those comments, please mark it as resolved for me, don't want to resolve something else by mistake.